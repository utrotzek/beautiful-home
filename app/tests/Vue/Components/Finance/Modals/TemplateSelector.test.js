import {shallowMount} from "@vue/test-utils";
import templateSelector from "../../../../../resources/js/components/Finance/Modals/TemplateSelector";
import moxios from "moxios";
import _ from "lodash";

import TemplateDataFixtures from "../Fixtures/TemplateFixtures";
describe("accounting and planning connection modal", () => {
    let wrapper = null;

    beforeEach(async () => {
        moxios.install(window.axios);
        wrapper = shallowMount(templateSelector,{
            propsData: {show: true}
        });

        //mock functions
        wrapper.vm.$refs.modalProgress.start = jest.fn();
        wrapper.vm.$refs.modalProgress.done = jest.fn();

        moxios.stubRequest("/api/finance/template", {
            status: 200,
            response: _.clone(TemplateDataFixtures)
        });
        await wrapper.vm.loadTemplates();
    });

    afterEach( () => {
        moxios.uninstall(window.axios);
    });

    it ("renders properly", async () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it ("can select a template", async () => {
        wrapper.find(".template-entry:first-child").trigger("click");
        expect(wrapper.emitted().templateSelected[0]).toEqual([TemplateDataFixtures[0]]);
    });

    it ("can be closed", async () => {
        wrapper.find(".close").trigger("click");
        expect(wrapper.emitted().close).toBeTruthy();
    });
});
