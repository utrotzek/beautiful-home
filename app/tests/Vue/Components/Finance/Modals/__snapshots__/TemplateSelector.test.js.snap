// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounting and planning connection modal renders properly 1`] = `
<div
  class="modal-wrapper"
>
  <progressbar-stub
    color="#f8f9fa"
    easing="linear"
    errorcolor="#f44336"
    height="3"
    maximum="97.5"
    minimum="0.8"
    speed="350"
    trickle="true"
    tricklespeed="250"
    zindex="9999"
  />
   
  <transition-stub
    name="modal"
  >
    <div
      class="modal-mask"
    >
      <div
        class="modal-wrapper"
      >
        <div
          class="modal-dialog"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h3
                class="modal-title"
              >
                
                                Template auswählen
                            
              </h3>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <p>
                Wählen Sie eine Vorlage aus. Die Daten werden anschließend in den aktuellen Monat übernommen.
              </p>
               
              <ul
                class="list-group"
              >
                <li
                  class="list-group-item template-entry"
                >
                  
                                    Default template (2 Einträge)
                                
                </li>
                <li
                  class="list-group-item template-entry"
                >
                  
                                    Another template (2 Einträge)
                                
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
